<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>PL/SQL tutorial 47: How to declare user-define exception using a EXCEPTION variable | Coder Coacher - Coaching Coders</title><meta content="PL/SQL tutorial 47: How to declare user-define exception using a EXCEPTION variable - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Manish-Sharma/">Manish Sharma</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>PL/SQL tutorial 47: How to declare user-define exception using a EXCEPTION variable</b></h2><h5 class="post__date">2017-01-03</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/lpExvX0qfag" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">what's up Internet welcome back once
again i am manish from
rebellionrider.com as in the
introduction to exception handling which
was the last tutorial we discussed that
there are three ways of declaring
user-defined exception in Oracle
database in to this PL sequel tutorial
we are going to explore the first way
and learn how to declare user-defined
exception using a variable of exception
data right so wise without wasting your
time let's get started the process of
declaring user-defined exception is
divided into three parts and these three
parts are first declare a variable of
exception data type this variable is
going to take all the burden on its
shoulders then we is the exception this
is the part where you tell the compiler
about the condition which will trigger
the exception and then comes the last
part which is handle the exception this
is the section where you specify what
will happen when the error which you
raised will trigger
so let's write the code demonstrating
all the above steps one by one like
always I will try to keep the code as
simple as possible so that you guys can
understand the concept clearly let's do
it then
here we have the declaration section of
the anonymous block which we are writing
for today's demonstration in this
declaration section we have four
variables among these four variables
first three are normal number data type
variables and the fourth one which is
e^x underscore div zero is the special
exception data type variable this
variable will become our user-defined
exception for this program now let's
move on and write the execution section
of this anonymous block before moving
ahead let's take a quick break and look
at what we just did here I raised an
error using the raised condition in
order to raise the error we use the
raised statement in which we first write
the keyword raised followed by the
variable of exception data type which is
e^x underscore divs 0 in our case raises
statement Inc is the normal flow of
execution of the code as soon as
compiler comes across arrays condition
it transfers the control over to the
exception handler in order to avoid
these unwanted switches during the
control flow of the program we use the
if condition using if condition we are
making sure that this error will come
into action
only when the divisor is equal to 0 now
let's move ahead and write the next
segment of this section
here is our next segment where we are
performing the automatic of division
operation and storing the result into
the variable var underscore reserve as
well as displaying back the output using
the DBMS output statement it's pretty
simple next we will write the exception
handler for the variable X underscore
Dave zero here we have the exception
handler for the variable X underscore
Dave zero in this exception handling
section we have a DBMS output statement
which will get displayed when our
user-defined error which is e^x
underscored if 0 occurs now before
executing this program let's take a
quick overview of it in this program we
declare a variable of exception datatype
and named it e^x underscore dave 0 this
variable is working as a user-defined
exception for our anonymous block this
PS equal user-defined exception will
come into action only when the divisor
is equal to 0 if the divisor is 0 as it
is in our case the error will be reached
and then the control of the program will
escape all the following statements and
look for the matching exception handler
if it finds any then it will perform the
action accordingly otherwise it will
either terminate the program or prompt
you with a system define error for
unhandled user-defined error for more
detailed information you can visit my
blog on the same topic link is in the
description for now let's execute this
PL sequel block and see what happens
so here is the output which is this
frame from the exception handler this
string is here because our divisor is 0
let's change the value of our divisor
and then try executing this program once
again
let's execute the code once again this
time we have the actual value as a
result which shows there is no error and
the program control has executed all the
statements that is how we declare user
define exception using a variable of
exception datatype how this example help
you with the concept and you learn
something new if you liked this video
then give it a thumbs up and do make
sure to share this PL sequel tutorial on
your social media as it will help others
in learning as well as me and my channel
that is the tutorial on how to declare
user defined exception using a variable
of exception data time thanks for
watching I will be back soon with the
next tutorial stay tuned this is money
from rebellionrider.com</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>